src/main/kotlin/no/nav/familie/ef/sak/økonomi/Utbetalingsoppdrag.kt

```diff
@@ -0,0 +1,90 @@
+package no.nav.familie.ef.sak.økonomi
+
+import no.nav.familie.ef.sak.økonomi.dto.AndelTilkjentYtelse
+import no.nav.familie.ef.sak.økonomi.dto.TilkjentYtelse
+import no.nav.familie.kontrakter.felles.oppdrag.Opphør
+import no.nav.familie.kontrakter.felles.oppdrag.Utbetalingsoppdrag
+import no.nav.familie.kontrakter.felles.oppdrag.Utbetalingsoppdrag.KodeEndring.NY
+import no.nav.familie.kontrakter.felles.oppdrag.Utbetalingsoppdrag.KodeEndring.UEND
+import no.nav.familie.kontrakter.felles.oppdrag.Utbetalingsperiode
+import no.nav.familie.kontrakter.felles.oppdrag.Utbetalingsperiode.SatsType.MND
+import no.nav.fpsak.tidsserie.LocalDateSegment
+import java.math.BigDecimal
+
+// Må forsikre oss om at tidslinjesegmentene er i samme rekkefølge for å få konsekvent periodeId
+// Sorter etter fraDato, sats, og evt til dato
+// PeriodeId = Vedtak.id * 1000 + offset
+// Beholder bare siste utbetalingsperiode hvis det er opphør.
+fun lagUtbetalingsoppdrag(saksbehandlerId: String,
+                          tilkjentYtelse: TilkjentYtelse,
+                          andelerTilkjentYtelse: List<AndelTilkjentYtelse>): Utbetalingsoppdrag {
+
+    val erOpphør = tilkjentYtelse.opphørFom!=null
+
+    val utbetalingsperiodeMal =
+            if (erOpphør)
+                UtbetalingsperiodeMal(tilkjentYtelse, true, tilkjentYtelse.forrigeTilkjentYtelseId!!)
+            else
+                UtbetalingsperiodeMal(tilkjentYtelse)
+
+    val tidslinjeMap = beregnUtbetalingsperioder(andelerTilkjentYtelse)
+
+    val utbetalingsperioder = tidslinjeMap.flatMap { (klassifisering, tidslinje) ->
+        tidslinje.toSegments()
+                .sortedWith(compareBy<LocalDateSegment<Int>>({ it.fom }, { it.value }, { it.tom }))
+                .mapIndexed { indeks, segment ->
+                    utbetalingsperiodeMal.lagPeriode(klassifisering, segment, indeks)
+                }.kunSisteHvis(erOpphør)
+    }
+
+    return Utbetalingsoppdrag(
+            saksbehandlerId = saksbehandlerId,
+            kodeEndring = if (!erOpphør) NY else UEND,
+            fagSystem = FAGSYSTEM,
+            saksnummer = tilkjentYtelse.saksnummer,
+            aktoer = tilkjentYtelse.personIdentifikator,
+            utbetalingsperiode = utbetalingsperioder
+    )
+}
+
+fun <T> List<T>.kunSisteHvis(kunSiste: Boolean): List<T> {
+    return this.foldRight(mutableListOf()) { element, resultat ->
+        if (resultat.size == 0 || !kunSiste) resultat.add(0, element);resultat
```

Comment:
Hva slags triks er det på slutten her `;resultat` 

For å slippe å skrive `return resultat` ? I så fall vil jeg slå et slag for `return resultat` for lesbarhet om 5 år :)

Reply:
Ikke noe triks. lambda-uttrykket ender om med verdi tilsvarende siste statement i eksekveeringen. Hvis man skriver return så gjelder det for metoden kunSisteHvis. 
Synes likevel at vi kan bruke linjeskift i steedet for semikolon.

Reply:
Aha, da tenker jeg linjeskift er ryddig iallefall.